### ISSUE TITLE
Cursor motion lag, Xbanish high CPU use

### ISSUE THREAD

**Issue by Iiridayn** [jcs/xbanish#63]
Arch Linux, XFCE, Xbanish 1.7, launched from `.xinitrc`.

Updated my system last night and rebooted (~330 packages, so difficult to say if the update is related). This morning came back to the computer, mouse cursor was very laggy and conky showed CPU spikes when I moved the cursor. `htop` showed xbanish was spiking CPU hard when the mouse moved. I killed it and performance returned to normal after the screen froze for a few seconds. Now running w/`-d` to see if anything comes up. Only thing that might have been unusual is I might have not moved the mouse after I finished using the computer last night before the monitors were blanked due to inactivity.

**Comment by jcs** [comment]
I don't know what I'm supposed to do with this

**Comment by Iiridayn** [comment]
Yeah, reports of intermittent bugs are a pain. I've got symptoms but no way to reproduce it, and no log output. If nothing else, it serves to indicate there is an uncommon edge case being handled poorly _somewhere_.

**Comment by jcs** [comment]
Reopen if you have some debugging information...

**Comment by Iiridayn** [comment]
```
using xinput2 raw motion events
attaching to keyboard device Virtual core XTEST keyboard (use 3)
attaching to keyboard device Power Button (use 3)
attaching to keyboard device Power Button (use 3)
attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
```
Second boot, same issue. This time getting this constantly repeated in the logs. Post-restart, working without massive lag, so this message might not be the cause though. Will modify my startup files to pipe to a log file.

**Comment by jcs** [comment]
That would indicate that the X server is constantly sending new `XDevicePresenceNotifyEvent` events which are supposed to happen only when devices are plugged in or unplugged.

**Comment by Iiridayn** [comment]
Okay, I have a 1.4G log file after running it for 12 minutes, just started hitting really high CPU use.

**Comment by jcs** [comment]
Does your /var/log/Xorg.0.log show any activity?

**Comment by Iiridayn** [comment]
```
$ sort xbanish.log | uniq -c | sort -r | head -n 10
38094300 mouse moved, already unhiding cursor
   2270 attaching to keyboard device Power Button (use 3)
   1135 using xinput2 raw motion events
   1135 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
   1135 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
   1135 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
   1135 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
   1135 attaching to keyboard device Virtual core XTEST keyboard (use 3)
   1135 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
    291 keystroke, already hiding cursor
```

Looks like it's really hyperactive about detecting mouse movements? Doesn't seem related to the hardware getting redetected constantly.

**Comment by Iiridayn** [comment]
Last hundred lines of the log file, after grouping by unique lines, so it should be during the high CPU usage phase:
```
$ uniq -c xbanish.log | tail -n 100
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
 626080 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
 355074 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
 169339 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
   3370 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
 372152 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
   7917 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
 331676 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
 538054 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
  19276 mouse moved, already unhiding cursor
      1 using xinput2 raw motion events
      1 attaching to keyboard device Virtual core XTEST keyboard (use 3)
      2 attaching to keyboard device Power Button (use 3)
      1 attaching to keyboard device Logitech Gaming Mouse G600 Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Keyboard (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ System Control (use 3)
      1 attaching to keyboard device ZSA Technology Labs ErgoDox EZ Consumer Control (use 3)
  47670 mouse moved, already unhiding cursor
      1 keystroke, hiding cursor
      1 mouse moved, unhiding cursor
  39675 mouse moved, already unhiding cursor
      1 mouse moved, alrea
```

**Comment by Iiridayn** [comment]
Looks like `/var/log/Xorg.0.log` is a closed session; file ends with `[2174870.625] (II) Server terminated successfully (0). Closing log file.`

**Comment by jcs** [comment]
Running with `-d` saving to a log file will probably cause high CPU just in itself since it logs every time the mouse is moved.

Does running `xinput test-xi2 --root` show similar events regarding device changes?

**Comment by Iiridayn** [comment]
Oddly, I didn't experience high CPU usage until around 10 minutes after I started running xbanish, despite continuing to use the computer as normal. Thus, I suggest that there is something more than just "debug mode is slow" going on in the log file I have.

I've installed `xinput`, and yes, around once a second there is a message printed looking vaguely like something getting detached and attached.

I would again emphasize that I had acceptable performance for around 10 minutes _despite_ those messages, which were constant before I started it with a log file, until it suddenly degraded, at which point I killed `xbanish`.
